### PR TITLE
fix: Include vertex_ai_beta in vertex_ai param mapping/Do not use google auth project_id

### PR DESCRIPTION
**Pull request by t968914**
## Title

Vertex_ai_beta chat completion does not work with provided project_id in model info because it is always overwritten by `project_id `returned from `google.auth.default()`. However, `google.auth.default()` returns None in cases where there is no service account. This will also not work when litellm is hosted in a seperate google project from vertex ai models. The project_id passed in as a param should take precedent to support this. 

## Type

🐛 Bug Fix

## Changes

- Include vertex_ai_beta when checking for vertex_ai provider type to map params
- Do not use google auth default credentials project_id

### CODE REVIEW

**Comment by ghost** [comment]
hey @t968914 should be fixed already with this - https://github.com/BerriAI/litellm/commit/6b14cf765708376490c5d88d3e54edc173c343b6

Waiting on a new release (v`1.40.31`). Should be live by EOD. 

Feel free to reopen, if that doesn't fix it.

**Comment by ghost** [comment]
Realized you had 2 changes here. Since we've addressed the default project issue, can you isolate the PR to 

> Include vertex_ai_beta when checking for vertex_ai provider type to map params

lgtm for merge

File: litellm/llms/vertex_httpx.py [hunk 1]
```diff
@@ -748,10 +748,12 @@ def load_auth(
             if project_id is None:
                 project_id = creds.project_id
         else:
-            creds, project_id = google_auth.default(
+            creds, creds_project_id = google_auth.default(
                 quota_project_id=project_id,
                 scopes=["https://www.googleapis.com/auth/cloud-platform"],
             )
+            if project_id is None:
+                project_id = creds_project_id
```
Comment (t968914) [inline litellm/llms/vertex_httpx.py:756]: Hi @krrishdholakia, I saw the change in 

> hey @t968914 should be fixed already with this - [6b14cf7](https://github.com/BerriAI/litellm/commit/6b14cf765708376490c5d88d3e54edc173c343b6)

This part is in the else block and for the application default credential case when a service account isn't being used. The google auth default method will always return None for project_id in this case, and similarly write over the previous value passed into this method.

Comment (t968914) [reply litellm/llms/vertex_httpx.py:756]: And thanks for taking a look at the PR

Comment (andrewmjc) [reply litellm/llms/vertex_httpx.py:756]: @krrishdholakia This is currently blocking us from being able to test vertex_ai_beta in local dev

Comment (t968914) [reply litellm/llms/vertex_httpx.py:756]: Our models in production are also deployed in a different GCP project than our litellm proxy, so we are unable to use vertex_ai_beta in production either

Comment (ghost) [reply litellm/llms/vertex_httpx.py:756]: noted - working on this today

Comment (ghost) [reply litellm/llms/vertex_httpx.py:756]: thanks for the review @andrewmjc @t968914

**Pull request merged**